compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -766,9 +766,11 @@ impl<'a, 'b> Context<'a, 'b> {
         // of each variable because we don't want to move out of the arguments
         // passed to this function.
         for (i, e) in self.args.into_iter().enumerate() {
-            for arg_ty in self.arg_unique_types[i].iter() {
-                args.push(Context::format_arg(self.ecx, self.macsp, e.span, arg_ty, i));
-            }
+            args.extend(
+                self.arg_unique_types[i]
+                    .iter()
+                    .map(|arg_ty| Context::format_arg(self.ecx, self.macsp, e.span, arg_ty, i)),
+            );
             heads.push(self.ecx.expr_addr_of(e.span, e));
         }
```

Comment:
If `self.args` and `heads` are of the same type (can't see that without type annotations) then since we're already doing `self.args.into_iter()`  this loop could be rewritten as `let heads = self.args.into_iter.map(|| ...).collect();`.

Reply:
Good idea. We could even `zip` `self.arg_unique_types`, but I don't suppose that it would provide a lot of gain, since even with `.iter().map()` the size should be known. And the code requires the numeric index, so the `enumerate` would have to stay there.

Btw, I force push the new changes, will it abort the perf run?

Reply:
No, perf keeps going since bors already started building